utils/getInvoices/tests/getInvoices.electricity.test.ts

```diff
@@ -6,143 +6,551 @@ import { ServiceType } from '@utils/constants'
+            price: 10,
+            sum: 10,
+          },
+        ],
+      }
+
+      const invoices = getInvoices({
+        service,
+        company,
+        prevPayment,
+      })
+
+      expect(invoices).toContainEqual({
+        type: ServiceType.Electricity,
+        amount: 0,
+        lastAmount: 0,
+        price: 10,
+        sum: 0,
+      })
+    })
   })
 
-  it('should NOT load Electricity without props', () => {
-    const company: Partial<IRealestate> = {}
-    const service: Partial<IService> = {}
-    const payment: Partial<IPayment> = {}
-    const prevPayment: Partial<IPayment> = {}
+  describe('props: { service, company, payment } with prio to payment', () => {
+    it('should NOT load when service = null, company = null, payment = null', () => {
+      const service: Partial<IService> = null
+      const company: Partial<IRealestate> = null
+      const payment: Partial<IPayment> = null
+
+      const invoices = getInvoices({
+        service,
+        company,
+        payment,
+      })
 
-    const invoices = getInvoices({
-      company,
-      service,
-      payment,
-      prevPayment,
+      expect(invoices).not.toContainEqual({
+        type: ServiceType.Electricity,
+      })
     })
+    it('should load when service = null, company = null, payment = { invoice: [Electricity] }', () => {
+      const service: Partial<IService> = null
+      const company: Partial<IRealestate> = null
+      const payment: Partial<IPayment> = {
+        invoice: [
+          {
+            type: ServiceType.Electricity,
+            amount: 12,
+            lastAmount: 1,
+            price: 10,
+            sum: 10,
+          },
+        ],
+      }
 
-    expect(invoices).not.toContainEqual(
-      expect.objectContaining({ type: ServiceType.Electricity })
-    )
+      const invoices = getInvoices({
+        service,
+        company,
+        payment,
+      })
+
+      expect(invoices).toContainEqual({
+        type: ServiceType.Electricity,
+        amount: 12,
+        lastAmount: 1,
+        price: 10,
+        sum: 10,
+      })
+    })
+    it('should load when service = { electricityPrice: 10 }, company = null, payment = { invoice: [Electricity] }', () => {
+      const service: Partial<IService> = {
+        electricityPrice: 10,
+      }
+      const company: Partial<IRealestate> = null
+      const payment: Partial<IPayment> = {
+        invoice: [
+          {
+            type: ServiceType.Electricity,
+            amount: 12,
+            lastAmount: 1,
+            price: 10,
+            sum: 10,
+          },
+        ],
+      }
+
+      const invoices = getInvoices({
+        service,
+        company,
+        payment,
+      })
+
+      expect(invoices).toContainEqual({
+        type: ServiceType.Electricity,
+        amount: 12,
+        lastAmount: 1,
+        price: 10,
+        sum: payment.invoice[0].sum || payment.invoice[0].price * (payment.invoice[0].amount - payment.invoice[0].lastAmount),
```

Comment:
Hard to read, use exact values instead. If we have `payment` argument provided the response shoul be completely the same, so `sum: 10` instead of `sum: payment.invoice[0].sum || payment.invoice[0].price * (payment.invoice[0].amount - payment.invoice[0].lastAmount)` is the case